dev-tools/composer.json

```diff
@@ -18,6 +18,9 @@
             "phpstan/extension-installer": true
         },
         "optimize-autoloader": true,
+        "platform": {
+            "php": "8.1"
+        },
```

Comment:
Why did you define `platform`? 🤔

Reply:
I matched here PHP that is min-supported by dev-tools/composer.json and same time used for SCA build.

I do not want to upgrade on my local machine any dependency to version that requires PHP 8.2 and put it to lock file, making it failing on SCA PHP 8.1 build

Reply:
So maybe put `8.1.*` in the `require`? Same effect, without faking the platform. And use Docker for Composer updates 😉.

Reply:
overkill

now, i can install&run on ^8.1 without need for docker to run on 8.1, same time I ensure i dont put 8.2-requiring dependency in lock file.

Reply:
Ok, but why 8.1? Can't we change CI definition to run dev-tools on 8.2?

Reply:
we can. but now we run on 8.1 in CI.
imho updating deps and changing runtime 2 different PRs.